app/Providers/AppServiceProvider.php

```diff
@@ -6,46 +6,44 @@
 
 use Acme\Academic\Repository\AcademicRepository;
 use Acme\Article\Repository\ArticleRepository;
-use App\Integration\Academic\Mapper\AcademicMapper;
-use App\Integration\Academic\Mapper\DatabaseAcademicMapper;
+use App\Integration\Academic\Mapper\FromAcademic\DefaultFromAcademicMapper;
+use App\Integration\Academic\Mapper\FromAcademic\FromAcademicMapper;
+use App\Integration\Academic\Mapper\FromArray\DefaultFromArrayMapper as FromArrayAcademicMapper;
+use App\Integration\Academic\Mapper\FromArray\FromArrayMapper as FromArrayAcademicMapperInterface;
 use App\Integration\Academic\Repository\AcademicQueryBuilderRepository;
-use App\Integration\Article\Mapper\ArticleMapper;
-use App\Integration\Article\Mapper\DatabaseArticleMapper;
+use App\Integration\Article\Mapper\FromArray\DefaultFromArrayMapper as FromArrayArticleMapper;
+use App\Integration\Article\Mapper\FromArray\FromArrayMapper as FromArrayArticleMapperInterface;
+use App\Integration\Article\Mapper\FromArticle\DefaultFromArticleMapper;
+use App\Integration\Article\Mapper\FromArticle\FromArticleMapper;
 use App\Integration\Article\Repository\ArticleQueryBuilderRepository;
 use App\Integration\Common\Query\CrudFacade;
 use App\Integration\Common\Query\CrudFacadeDefaultBuilder;
+use Illuminate\Log\Logger;
 use Illuminate\Support\ServiceProvider;
 
 class AppServiceProvider extends ServiceProvider
 {
```

Comment:
It's becoming too big. We have to split in different providers, one for every domain, and then include them or inside app.php or inside AppServiceProvider

Reply:
ATM there are  50 lines, I don't see any benefit.

Reply:
Separation of concerns. Think about the possibilities to split each domain in a composer library.

Reply:
Feel free to add more providers in a PR 😄